src/imap.rs

```diff
@@ -2321,22 +2342,23 @@ fn build_sequence_sets(mut uids: Vec<u32>) -> Vec<String> {
     }
 
     // Second, sort the uids into uid sets that are each below ~1000 characters
-    let mut result = vec![String::new()];
+    let mut result = vec![(Vec::new(), String::new())];
     for range in ranges {
-        if let Some(last) = result.last_mut() {
```

Comment:
I think better not to write code this way, `result` can't be empty by design, and if we it's difficult to rewrite the code in a way the compiler would know that, better to return an error (panics also don't seem good here since they are rather for terminating a program entered an irrecoverable state). But i don't want to clutter up the code with handling of such impossible situations and just wrote `context(0)`, but maybe we have an idiomatic way for that?

Reply:
Usually we would write `.context("result was empty")` or so in this case.

Reply:
Ok, will keep in mind. But rewritten the code to get rid of unnecessary error handling